core/src/page.rs

```diff
@@ -202,7 +202,8 @@ impl<'a> PageSetCursor<'a> {
                 CursorLocation::Root => ROOT_PAGE_ID,
                 CursorLocation::Page(p) => {
                     let child_page_idx = last_page_path(&self.path, self.depth).load_be::<u8>();
-                    child_page_id(*p.id(), child_page_idx)
+                    // extracted child_page_idx will never be bigger than 6 bits
+                    child_page_id(*p.id(), child_page_idx).unwrap()
```

Comment:
unwrap should never appear in production code if we can help it. use `expect` with a proof